dbr quant: clean up key types in AutoQuantizationState mappings

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #68432 dbr quant overhead[9/x]: precalculate when to skip op_convert_after_hook
* #68431 dbr quant: remove more asserts from hot paths
* #68404 speed up quantized relu6 inplace kernel
* #68374 dbr quant: simplify relatedness logic
* #68373 dbr quant overhead[8/x]: small speedup in op_needs_quantization
* #68372 dbr quant overhead[7/x]: speed up AutoQuantizationState.reset_to_new_call
* #68371 dbr quant overhead[6/x]: remove unneeded isinstance checks in `op_convert_before_hook`
* #68370 dbr quant overhead[5/x]: remove unnecessary asserts
* **#68369 dbr quant: clean up key types in AutoQuantizationState mappings**
* #68351 dbr quant overhead[4/x]: speed up hook type calculations
* #68350 dbr quant overhead[3/x]: speed up AutoQuantizationState.mark_cur_op_complete

Summary:

`AutoQuantizationState` has various mappings keyed on IDs. Only
`tensor_id_to_observer` actually needs string keys because it is an
`torch.nn.ModuleDict`.  This PR changes the other mappings to have
integer keys, for simplicity and performance.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR
```

Differential Revision: [D32463765](https://our.internmc.facebook.com/intern/diff/D32463765)

vkuzo committed:
dbr quant: clean up key types in AutoQuantizationState mappings

Summary:

`AutoQuantizationState` has various mappings keyed on IDs. Only
`tensor_id_to_observer` actually needs string keys because it is an
`torch.nn.ModuleDict`.  This PR changes the other mappings to have
integer keys, for simplicity and performance.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR
```

[ghstack-poisoned]

vkuzo committed:
Update on "dbr quant: clean up key types in AutoQuantizationState mappings"


Summary:

`AutoQuantizationState` has various mappings keyed on IDs. Only
`tensor_id_to_observer` actually needs string keys because it is an
`torch.nn.ModuleDict`.  This PR changes the other mappings to have
integer keys, for simplicity and performance.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR
```

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32463765).

jerryzh168 commented on torch/ao/quantization/_dbr/model_utils.py:
```diff
@@ -103,7 +103,9 @@ def attach_scale_zp_values_to_model(
         for tensor_id, observer in qstate.tensor_id_to_observer.items():
             scale, zp = observer.calculate_qparams()
-            qstate.tensor_id_to_scale_zp[tensor_id] = (scale, zp)
+            # tensor_id_to_observer is a ModuleDict which has to have string keys
```
thanks for the comment

vkuzo committed:
Update on "dbr quant: clean up key types in AutoQuantizationState mappings"


Summary:

`AutoQuantizationState` has various mappings keyed on IDs. Only
`tensor_id_to_observer` actually needs string keys because it is an
`torch.nn.ModuleDict`.  This PR changes the other mappings to have
integer keys, for simplicity and performance.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR
```

Differential Revision: [D32463765](https://our.internmc.facebook.com/intern/diff/D32463765)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32463765).

vkuzo committed:
Update on "dbr quant: clean up key types in AutoQuantizationState mappings"


Summary:

`AutoQuantizationState` has various mappings keyed on IDs. Only
`tensor_id_to_observer` actually needs string keys because it is an
`torch.nn.ModuleDict`.  This PR changes the other mappings to have
integer keys, for simplicity and performance.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR
```

Differential Revision: [D32463765](https://our.internmc.facebook.com/intern/diff/D32463765)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32463765).

vkuzo committed:
Update on "dbr quant: clean up key types in AutoQuantizationState mappings"


Summary:

`AutoQuantizationState` has various mappings keyed on IDs. Only
`tensor_id_to_observer` actually needs string keys because it is an
`torch.nn.ModuleDict`.  This PR changes the other mappings to have
integer keys, for simplicity and performance.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR
```

Differential Revision: [D32463765](https://our.internmc.facebook.com/intern/diff/D32463765)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32463765).

vkuzo committed:
Update on "dbr quant: clean up key types in AutoQuantizationState mappings"


Summary:

`AutoQuantizationState` has various mappings keyed on IDs. Only
`tensor_id_to_observer` actually needs string keys because it is an
`torch.nn.ModuleDict`.  This PR changes the other mappings to have
integer keys, for simplicity and performance.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR
```

Differential Revision: [D32463765](https://our.internmc.facebook.com/intern/diff/D32463765)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32463765).

vkuzo committed:
Update on "dbr quant: clean up key types in AutoQuantizationState mappings"


Summary:

`AutoQuantizationState` has various mappings keyed on IDs. Only
`tensor_id_to_observer` actually needs string keys because it is an
`torch.nn.ModuleDict`.  This PR changes the other mappings to have
integer keys, for simplicity and performance.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR
```

Differential Revision: [D32463765](https://our.internmc.facebook.com/intern/diff/D32463765)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32463765).
